### ISSUE TITLE
Parsing dates with format "yyyy.MM.dd (DDD) HH:mm:ss"

### ISSUE THREAD

**Issue by thoism** [HowardHinnant/date#467]
Hi,

Is there a way to parse dates with this format:

`"yyyy.MM.dd (DDD) HH:mm:ss" `

Thank you in advance

**Comment by HowardHinnant** [comment]
What is "(DDD)"?

**Comment by thoism** [comment]
It seems to be the day of the year. if I am not mistaken based on your documentation`%j`

**Comment by thoism** [comment]
```
    date::local_seconds tp ;
    if(in.fail() || in.bad()) {
      throw std::runtime_error(fmt::format("Failed to parse {}", time_str));
    }
    in >> parse("%Y.%m.%d (%j) %T", tp);
    auto timestamp = tp.time_since_epoch().count() // I guess this is the timestamp?
```

**Comment by thoism** [comment]
That's it man thanks!!!

**Comment by HowardHinnant** [comment]
Yes, I was just about to post something very close to what you just posted.  Checking `in.fail()` is sufficient.  You could also set `in.exceptions(ios::failbit)` and a failure to parse would then throw without you having to check for `in.fail()`.

Note that this will fail if the number parsed for %j is not consistent with the date parsed for %Y.%m.%d.

I recommend parsing into a `date::sys_seconds` instead of a `date::local_seconds`.  There is no difference functionally.  But a `date::sys_seconds` represents a timepoint in UTC (i.e. a Unix Time).  A `date::local_seconds` represents a local time that has an unspecified time zone.  Both get the same answer.  Just `date::sys_seconds` tells readers of the code what you're after.

Glad this library could help!
